app/Http/Resources/GalleryConfigs/SettingsConfig.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/**
+ * SPDX-License-Identifier: MIT
+ * Copyright (c) 2017-2018 Tobias Reich
+ * Copyright (c) 2018-2025 LycheeOrg.
+ */
+
+namespace App\Http\Resources\GalleryConfigs;
+
+use App\Models\Configs;
+use Spatie\LaravelData\Data;
+use Spatie\TypeScriptTransformer\Attributes\TypeScript;
+
+#[TypeScript()]
+class SettingsConfig extends Data
+{
+	public bool $default_old_settings;
```

Comment:
I'm not sure what the parent class `Data` do but, why not private properties with getters & setters?

Reply:
https://spatie.be/docs/laravel-data/v4/introduction the Data class transforms any public properties into JSON attributes at serialisation. What we could consider is making them read only (but didn't feel like experimenting :) )